### DISCUSSION TITLE
How to configure the tasks.json problemMatcher property?

### DISCUSSION THREAD

**Discussion by stijnherreman** [microsoft/DacFx#189] [Q&A] [answered]
I'm trying out the Microsoft.Build.Sql SDK for some existing databases, using the SQL Database Projects extension in VS Code. The build output gives me over 1000 warnings, so I tried configuring tasks.json with a problem matcher to get a better overview of the problems.

```json
{
	"version": "2.0.0",
	"tasks": [
		{
			"label": "build example",
			"type": "process",
			"command": "dotnet",
			"args": [
				"build",
				"${workspaceFolder}/example/example.sqlproj",
				"/property:GenerateFullPaths=true",
				"/consoleloggerparameters:NoSummary"
			],
			"problemMatcher": "$msCompile",
			"group": {
				"kind": "build",
				"isDefault": true
			}
		}
	]
}

```

Unfortunately `$msCompile` doesn't pick up these warnings. Is there a problem matcher I can refer to, or a regular expression I can copy-paste?

**Comment by yooakim** [comment]
I just tested and for me this works in VS Code, I get the problems listed in the "Problems tab. I know this is a very old post but maybe worth trying @stijnherreman ?

This is using `<Sdk Name="Microsoft.Build.Sql" Version="2.1.0" />` to build.

```json
    {
      "label": "Build",
      "type": "shell",
      "command": "dotnet",
      "args": [
        "build",
        "--configuration",
        "Release",
        "--nologo",
        "example.sqlproj",
        "/p:RunSqlCodeAnalysis=true",
        "/p:WarningsAsErrors=\"\"",
        "/property:GenerateFullPaths=true",
				"/consoleloggerparameters:NoSummary"
      ],
      "options": {
        "cwd": "${workspaceFolder}"
      },
      "problemMatcher": "$msCompile",
      "group": {
        "kind": "build",
        "isDefault": true
      },
      "detail": "Builds the SQL project (Release, SQL code analysis)"
    }
```

**Reply by dzsquared** [reply, marked as answer]
we've introduced the problem matching directly in the sqlproj extension in vscode - https://github.com/microsoft/vscode-mssql/blob/7d4df3232ba9b1f35d042cb3384494f368737d51/extensions/sql-database-projects/package.json#L545

and @yooakim is right on with the tasks json definition (which can also be setup by the sqlproj extension during project creation, but only manually after the project is created)

**Reply by yooakim** [reply]
@dzsquared So should I use this in my `tasks.json`:
```json
      "problemMatcher": "$sqlproj-problem-matcher",
```
or the more common:
```json
      "problemMatcher": "$msCompile",
```

**Reply by ErikEJ** [reply]
Looks like it is "problemMatcher": "$sqlproj-problem-matcher",

(New project in latest extension version)
